### ISSUE TITLE
EXCLUDE_SYMBOLS incorrect behavior in referenced list

### ISSUE THREAD

**Issue by dima-lech** [doxygen/doxygen#8061]
When adding an _EXCLUDE_SYMBOLS_ entry, the references list is generated incorrectly in HTML documentation.
See following example:

```
void a()
{
    b();
    c();
}

void b()
{
    c();
}

void c()
{
    
}
```

**Without _EXCLUDE_SYMBOLS_:**
Output is as expected
> a()
> void a    (       )   
> References b(), and c().
> 
> b()
> void b    (       )   
> References c().
> 
> c()
> void c    (       )

**Actual output with _EXCLUDE_SYMBOLS = b_:**
`c()` is missing from `a()` references list
> a()
> void a    (       )   
> 
> c()
> void c    (       )

**Expected output with _EXCLUDE_SYMBOLS = b_:**
> a()
> void a    (       )   
> **References c().**
> 
> c()
> void c    (       )

**Version**
1.8.20 (Windows 10 64-bit)

**Comment by albert-github** [comment]
First investigation lead to that the problem surfaced in version 1.8.12, the last good version was:
```
Commit: 968ad0f144af6a7d411e1e909b0dc8d76472f363 [968ad0f]
Date: Saturday, March 19, 2016 4:34:33 PM
Merge branch 'master' of github.com:doxygen/doxygen
```
the next commit shows the problem
```
Commit: 9abcad810b8d41d338d501ff5b32524e1ced7f33 [9abcad8]
Date: Sunday, March 20, 2016 11:51:25 AM
Bug 762670 - References for one function can inherit References from subsequent non documented function
```

Looking at the code without the `EXCLUDE_SYMBOLS = b` we see:
![aa_8c_a2edadf23dba309e5902af6c6dc829482_cgraph](https://user-images.githubusercontent.com/5223533/94369264-03703f80-00e9-11eb-9dd8-ed803cdda001.png)

i.e. there is a "direct connection" from `a` to `c` and an "indirect connection" through `b`. With the `EXCLUDE_SYMBOLS = b` we would expect, like it is in 1.8.11 that "direct connection" remains but that only the "indirect connection" disappears.

When we add a function `d` that is only direct called from `a` we would expect that with `EXCLUDE_SYMBOLS = b` this connection is shown but due to the commit  9abcad810b8d41d338d501ff5b32524e1ced7f33  also this connection is gone.

**Comment by albert-github** [comment]
Converting the problem to Fortran (bb.f) like:
```
      subroutine a1()
          call b1()
          call c1()
      end

      subroutine b1()
          call c1()
      end

      subroutine c1()
      end
```
or converting it to python (cc.py) like:
```
def a2():
    b2()
    c2()

def b2():
    c2()

def c2():
    pass
```
we see that even after the commit the result remains as expected.

**Comment by albert-github** [comment]
I wanted to see if there would be an "easy" fix for this problem so I changed the line:
```
    else // found member, but it is not linkable, so make sure content inside is not assigned
```
to
```
    else if (!md->isFunction())// found member, but it is not linkable, so make sure content inside is not assigned
```
for the problem in this issue it looked OK. So I also tried:
```
int a1;
int b1;
int c1;


void a()
{
    b();
    c();
    b1++;
    c1++;
}

void b()
{
    c();
    c1++;
}

void c()
{

}
```
but this gave problems so I tried (again missing information):
```
    else if (0)// found member, but it is not linkable, so make sure content inside is not assigned
```
it looked all OK, but I thought this cannot be, would undo the complete fix for #5958 

So I had a go at the original problem and noticed that the "Module" tab was gone in the current version (compared to 1.8.12. It was gone since 1.8.15), so I had a deeper look at the output and especially at the results presented in `html/doxygen__references__test_8c.html`.
Till and including version 1.8.14 we see at the top of this file:
![file_1_8_14](https://user-images.githubusercontent.com/5223533/94462404-c0cb6780-01bb-11eb-9d3e-175580cb7f5e.jpg)

from 1.8.15 onward we see here:
![file_1_8_15](https://user-images.githubusercontent.com/5223533/94462440-ce80ed00-01bb-11eb-9a85-169c14a60f2d.jpg)

in other words all entities are dropped under "classes" even macros and variables etc.

Looks like there are some deeper problems with this example...

**Comment by albert-github** [comment]
For the second problem in the issue "Call Graph corrupt and/or incomplete" (#9096) I've just pushed a proposed patch, pull request #9396 and this also solves the issue as reported here.

**Comment by albert-github** [comment]
The pull request #9096 for the call graph problem has been integrated into the master version
